sycl/CMakeLists.txt

```diff
@@ -198,6 +198,8 @@ configure_file("${version_header}.in" "${version_header}")
 # Copy SYCL headers
 add_custom_target(sycl-headers ALL
 COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/CL ${SYCL_INCLUDE_BUILD_DIR}/CL
+COMMAND ${CMAKE_COMMAND} -E copy_directory ${sycl_inc_dir}/sycl ${SYCL_INCLUDE_BUILD_DIR}/sycl
+COMMAND ${CMAKE_COMMAND} -E copy ${sycl_inc_dir}/sycl_esimd.hpp ${SYCL_INCLUDE_BUILD_DIR}/sycl_esimd.hpp
```

Comment:
@kbobrovs, do you know why ESIMD library headers are put separate from other Intel extensions: https://github.com/intel/llvm/tree/sycl/sycl/include/CL/sycl/intel?
I think if we move them there, cmake changes won't be required.

Reply:
@bader, because it is not inside the cl namespace, that was agreed upon decision to have sycl::intel::gpu as the namespace

Reply:
sycl/include/sycl_esimd.hpp has only defines.
All other headers declare symbols in `cl` namespace.
I don't see any reason to make exceptions for ESIMD extension.

Reply:
> All other headers declare symbols in cl namespace.

@bader, You are right. This wasn't the original intent, as initially it was requested to have this library outside the cl:: namespace, so I'm little surprised with presence of `__INLINE_NAMESPACE(cl)` in ESIMD headers.

But since now we consider this as a usual vendor extension, `__INLINE_NAMESPACE(cl)` and the full namespace `cl::sycl::intel::gpu` seems correct and according with other extensions, such as `sub_group`. So we move all the headers under `include/CL/sycl/intel/esimd` directory, and the sycl_esimd.h - to `include/CL/sycl/intel` - do you agree?

@rolandschulz, @Pennycook - FYI, please comment if you have other thoughts.

Reply:
This sounds reasonable to me. This should simplify the maintenance of this extension.

Tagging @gmlueck.

Reply:
Moved ESIMD extension to `include/CL/sycl/intel` as noted in previous comments.

Reply:
Moved ESIMD extension to `include/CL/sycl/intel` as noted in previous comments.

Reply:
Thanks. LGTM.

Reply:
> But since now we consider this as a usual vendor extension, __INLINE_NAMESPACE(cl) and the full namespace cl::sycl::intel::gpu seems correct and according with other extensions, such as sub_group. So we move all the headers under include/CL/sycl/intel/esimd directory, and the sycl_esimd.h - to include/CL/sycl/intel - do you agree?

Catching up on email after vacation.  Actually, according to SYCL 2020 provisional, vendor defined extensions should be in the `sycl::ext::<vendorname>` namespace, so `sycl::ext::intel::gpu` would the right spelling.

Likewise, vendor extension header files should have a path prefix `SYCL/ext/<vendorname>`, so this header should probably be under `include/CL/sycl/ext/intel`.

Reply:
We don't yet implement SYCL 2020, also existing extensions don't follow the `sycl::ext::<vendorname>` neither - see `sycl::intel::sub_group` for example. So my suggestion is to use `sycl::intel::gpu` and then update all extensions when moving to SYCL 2020 if there will be such decision for the extensions which will have existed prior to new spec adoption.